tests/main/validate-container-happy/task.yaml

```diff
@@ -0,0 +1,37 @@
+summary: check the symlinks following the right track
+
+environment:
+    SNAP: test-snapd-validate-container-happy
+    SNAP_INSTALL_DIR: $(pwd)/snap-install-dir
+    SNAP_MOUNT_DIR: $(os.paths snap-mount-dir)
+
+prepare: |
+    snap pack test-snapd-validate-container-happy
+
+execute: |
+
+    # We shouldn't use relative symlinks in Github as they cannot be packed correctly.
```

Comment:
Can you unpack the symlinks before running snap try? The main code that is being exercised is the code that validates the container/skeleton of the snap is correct which only runs initially when we do snap pack and snap try, so here when you run snap try before the symlinks have been extracted nothing is being checked. 

Additionally, if you could check that the symlinks are working after running snap try that would be good to check as well, but that is not as critical as the check after we install the packed snap since snap try is really just doing a bind mount.

Reply:
Switched to this structure:

1. Untar the symlinks
2. "try" and check if symlinks exist
3. "pack" and "install" the snap
4. check to see if the symlinks are pointing to the right targets